### ISSUE TITLE
Some strings cause an infinite loop

### ISSUE THREAD

**Issue by maxpereira** [discordier/sam#16]
When a string is inputted where a word or set of characters is repeated with no spaces it causes an infinite loop where nothing will be spoken, the window will hang, and the developer console will spam output. Tested on the most current version of both Firefox and Chrome.

Steps to replicate:
1. Go to https://discordier.github.io/sam/
2. Type "`SoftwareSoftwareSoftwareSoftware`" in the text box
3. Press "Say" button

Other strings that produce same results:
`dhaaaxaxaxaxaxaxaxaxaxaxaxaxax`
`CurrentlyCurrentlyCurrentlyCurrently`

**Comment by Trollgefine** [comment]
My sam text to speech is not working￼

**Comment by tyomitch** [comment]
@maxpereira this was an intentional limitation in the C64 SAM, faithfully inherited by the JS port.
[The manual](http://www.retrobits.net/atari/sam.shtml#ch4.0) says: 
> S.A.M. is capable of speaking only 2.5 seconds of speech without a break (this is the size of his "breath"). If the string to be spoken exceeds this, S.A.M. will insert short breaks every 2.5 seconds. [...] S.A.M. uses the spaces between words to makes his sentence-breaking decisions. If a single word requires more than 2.5 seconds to say, S.A.M. will not be able to insert his own breaks and will therefore be unable to say the word.

@discordier you may want to re-tag it as a feature request -- it's entirely valid and sensible, but it's not a bug :)
I have a downstream fix which will require some work to shape it into a PR...

**Comment by discordier** [comment]
You are right @tyomitch. I reflagged it but think we should alter this behaviour.
IMO this was only introduced to keep the buffer size of original sam small.
In the end, we should at least break if we encounter such a condition instead of freezing the browser.
